### PR TITLE
feat(core): add track events for cmdk

### PR DESCRIPTION
**Pull request by JimmFly**

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#7668** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7668?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7667** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7667?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7664** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7664?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7661** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7661?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7735** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7735?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7695** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7695?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/commands/affine-creation.tsx [hunk 1]
```diff
@@ -4,16 +4,19 @@ import type { createStore } from 'jotai';
...
 export function registerAffineCreationCommands({
   store,
   pageHelper,
   t,
+  moduleName,
 }: {
   t: ReturnType<typeof useI18n>;
   store: ReturnType<typeof createStore>;
   pageHelper: ReturnType<typeof usePageHelper>;
+  moduleName: string;
```
Comment (EYHN) [inline packages/frontend/core/src/commands/affine-creation.tsx]: here are several issues

moduleName doesn't seem to update responsively
registering so many commands has a significant impact on performance, so it is not recommended to add dependencies.

I think we can just delete this and I'll implement it in another way later.

Comment (JimmFly) [reply packages/frontend/core/src/commands/affine-creation.tsx]: Should I delete this event tracking directly or remove the "page, segment, module" attributes?

File: packages/frontend/core/src/commands/affine-creation.tsx [hunk 2]
```diff
@@ -29,6 +30,8 @@ export function registerAffineCreationCommands({
           }
         : undefined,
       run() {
+        track.$.cmdk.creation.createDoc({ control: 'newPage' });
```
Comment (forehalo) [inline packages/frontend/core/src/commands/affine-creation.tsx]: { type: 'page' | 'edgeless' }

File: packages/frontend/core/src/commands/affine-creation.tsx [hunk 3]
```diff
@@ -53,6 +60,10 @@ export function registerAffineCreationCommands({
       icon: <PlusIcon />,
       label: t['com.affine.cmdk.affine.new-workspace'](),
       run() {
+        track.$.cmdk.creation.createWorkspace({
+          control: 'newWorkspace',
```
Comment (forehalo) [inline packages/frontend/core/src/commands/affine-creation.tsx:64]: don't have to specify control, the segment/module matrix can identify this event already

File: packages/frontend/core/src/commands/affine-creation.tsx [hunk 4]
```diff
@@ -67,6 +78,10 @@ export function registerAffineCreationCommands({
         return environment.isDesktop;
       },
       run() {
+        track.$.cmdk.creation.createWorkspace({
+          control: 'importWorkspace',
```
Comment (forehalo) [inline packages/frontend/core/src/commands/affine-creation.tsx]: control: 'import'

File: packages/frontend/core/src/commands/affine-help.tsx [hunk 5]
```diff
@@ -21,6 +22,7 @@ export function registerAffineHelpCommands({
       icon: <NewIcon />,
       label: t['com.affine.cmdk.affine.whats-new'](),
       run() {
+        track.$.cmdk.help.whatNew();
```
Comment (forehalo) [inline packages/frontend/core/src/commands/affine-help.tsx]: showWhatsNew

File: packages/frontend/core/src/commands/affine-navigation.tsx [hunk 6]
```diff
@@ -28,6 +28,10 @@ export function registerAffineNavigationCommands({
       icon: <ArrowRightBigIcon />,
       label: t['com.affine.cmdk.affine.navigation.goto-all-pages'](),
       run() {
+        track.$.cmdk.navigation.goTo({
```
Comment (forehalo) [inline packages/frontend/core/src/commands/affine-navigation.tsx]: navigate({ to: 'allDocs' })

File: packages/frontend/core/src/commands/affine-settings.tsx [hunk 7]
```diff
@@ -34,6 +35,9 @@ export function registerAffineSettingsCommands({
       icon: <SettingsIcon />,
       preconditionStrategy: () => theme.theme !== 'system',
       run() {
+        track.$.cmdk.settings.changeColorMode({
```
Comment (forehalo) [inline packages/frontend/core/src/commands/affine-settings.tsx]: changeAppSetting({ key: '', value: ''  })

File: packages/frontend/core/src/hooks/affine/use-register-blocksuite-editor-commands.tsx [hunk 8]
```diff
@@ -102,6 +102,8 @@ export function useRegisterBlocksuiteEditorCommands() {
         icon: mode === 'page' ? <PageIcon /> : <EdgelessIcon />,
         label: t['com.affine.page-properties.page-info.view'](),
         run() {
+          track.$.cmdk.editor.viewInfo();
```
Comment (forehalo) [inline packages/frontend/core/src/hooks/affine/use-register-blocksuite-editor-commands.tsx]: `track.$.cmdk.docInfo.open()`

File: packages/frontend/core/src/hooks/affine/use-register-blocksuite-editor-commands.tsx [hunk 9]
```diff
@@ -141,6 +145,8 @@ export function useRegisterBlocksuiteEditorCommands() {
             : t['com.affine.pageMode.page']()
         }`,
         run() {
+          track.$.cmdk.editor.switchPageMode();
```
Comment (forehalo) [inline packages/frontend/core/src/hooks/affine/use-register-blocksuite-editor-commands.tsx]: `{ mode }`

File: packages/frontend/core/src/mixpanel/events.ts [hunk 10]
```diff
@@ -3,7 +3,36 @@
...
+  | 'export'
+  | 'moveToTrash'
+  | 'restoreFromTrash'
+  | 'revealDocHistoryModal'
+  | 'save'
+  | 'restartToUpgrade'
+  | 'toggleSidebar'
+  | 'whatNew'
+  | 'contactUs'
+  | 'openChangelog';
```
Comment (forehalo) [inline packages/frontend/core/src/mixpanel/events.ts]: control the amount of events and reuse them if possible, for example all setting changes and reuse `changeAppSetting()`

File: packages/frontend/core/src/commands/affine-creation.tsx [hunk 11]
```diff
@@ -53,6 +60,10 @@ export function registerAffineCreationCommands({
       icon: <PlusIcon />,
       label: t['com.affine.cmdk.affine.new-workspace'](),
       run() {
+        track.$.cmdk.creation.createWorkspace({
```
Comment (forehalo) [inline packages/frontend/core/src/commands/affine-creation.tsx]: `workspace.createWorkspace`

File: packages/frontend/core/src/mixpanel/events.ts (hunk 10 above)
Comment (JimmFly) [reply packages/frontend/core/src/mixpanel/events.ts]: I sorted out the event types in this PR and deleted some duplicate events.

File: packages/frontend/core/src/mixpanel/events.ts [hunk 12]
```diff
@@ -2,21 +2,26 @@
...
 type AppEvents =
   | 'checkUpdates'
   | 'downloadUpdate'
   | 'downloadApp'
   | 'quitAndInstall'
   | 'openChangelog'
-  | 'dismissChangelog';
+  | 'dismissChangelog'
+  | 'contactUs'
+  | 'toggleSidebar'
```
Comment (forehalo) [inline packages/frontend/core/src/mixpanel/events.ts]: reuse with `toggle`, `nativationPanel.$.toggle()`

**Review by forehalo — APPROVED** [review]

**Pull request merged**